src/Plotting/StrengthEnvelope.jl

```diff
@@ -0,0 +1,319 @@
+using Parameters
+using SpecialFunctions: erfc
+using .GLMakie
+
+export StrengthEnvelope, ConstantTemp, LinearTemp, HalfspaceCoolingTemp, GP_Compute_ThermalStructure, LithPres
+
+abstract type AbstractThermalStructure end  
+
+"""
+    ConstantTemp(T=1000)
+    
+Sets a constant temperature inside the box
+Parameters
+===
+- T : the value
+"""
+@with_kw_noshow mutable struct ConstantTemp <: AbstractThermalStructure
```

Comment:
Does this really need to be mutable? If this number really changes it's probably better to make a normal immutable struct and just reallocate it doing `ConstantTemp(T)`